Userland/Libraries/LibGUI/FileSystemModel.cpp

```diff
@@ -718,17 +718,24 @@ bool FileSystemModel::fetch_thumbnail_for(Node const& node)
                 strong_this->m_thumbnail_progress_total = 0;
             }
 
-            if (with_success)
+            if (with_success && (!strong_this->m_ui_update_timer.is_valid() || strong_this->m_ui_update_timer.elapsed_time() > 50_ms)) {
```

Comment:
We've got 50ms here, but the commit message says 100ms?

Reply:
Oops, I've set it to 100ms; that's what I used in the video and I couldn't see a massive difference between 50 and 100ms. That number is kind of pulled out of the air though, so happy to change it if there is a more principled value.